planner/core/logical_plan_builder.go

```diff
@@ -4516,6 +4530,10 @@ func CheckUpdateList(assignFlags []int, updt *Update) error {
 	return nil
 }
 
+func isCTE(tl *ast.TableName) bool {
```

Comment:
Can we move it to parser?

Reply:
I think the judgement `isCTE` is not general. There can be used in a local scope. We've better add a comment so that other people do not misuse it.

Reply:
@winoros a comment for this function is added, PTAL